cumulusci.yml

```diff
@@ -441,6 +451,23 @@ flows:
                 options:
                     unmanaged: False
 
+    preferred_email_testing:
+        description: Deploys Preferred Email configuration and runs apex tests. 
+        steps:
+            1: 
+                task: deploy_preferred_email_config
+
+            2:
+                task: run_tests
+                options: 
+                    test_name_match: '%_MTST'
```

Comment:
Have y'all confirmed hands on that this works with CumulusCI 3.23.0? I think you'll need an explicit option here to mark that we are looking for unmanaged tests.

I also just want to raise that unmanaged Apex tests are a really, really unusual pattern. Have all alternatives been exhausted? Right now, it looks like these are only queued up to run during regression, not during any feature or beta builds. That seems risky to me. How/where is the functionality that is being tested being delivered to customers?

Reply:
Thanks @davidmreed I updated CCI and updated yml to be explicit for unmanaged tests. 

We wanted to replicate a few scenarios in the functionality where customers would have their own fields working with EDA's functionality. Our managed tests cover all the other scenarios. This is the PR for the original functionality https://github.com/SalesforceFoundation/EDA/pull/1359

Reply:
Thanks. I am still concerned about the fact that these Apex tests are currently queued up to run only during the build of a regression org. Failures in that context will not be very visible, and they're neither run nor tracked in MetaCI, which increases the risk of an actual regression or our missing critical information.

What would you think about these two possibilities for covering this testing expectation?

1. Run the tests in a separate Feature Test build that would be required to pass on every PR. I don't love this idea because it requires us to run a new, parallel build just to ensure that these tests run and have their output captured, but I like the visibility it provides.
2. (My preference) Cover the testing gap by deploying the supporting metadata as part of both `config_qa` and `config_regression`, and execute functional tests via Robot. That doesn't require a new build and does ensure that you'll get a visible failure early if there's a regression in this functionality.

(cc'ing other likely interested parties: @jofsky, @jstvz , @MatthewBlanski).

Reply:
@davidmreed For this Work Item, we scoped it to only run on Regression Org. We would like these to run on every build eventually. We're laser-focused on replicating real-world scenarios in a fashion that is dev-maintainable and gives meaningful asserts as we have dynamic apex that is divergent based on managed vs unmanaged fields.

The benefits of having it as a unit test instead of robot are:
1. Anyone with Salesforce knowledge can run it in an org the tests are deployed into with meaningful asserts, meaning developers can get on top of it immediately.
2. This is maintainable with Apex knowledge only, meaning this is a dev-accepted burden and something we would write tests for if the platform was capable of it.
3. It obviates the inherent instability of interacting with the front-end.

I'd love to have a conversation about this to explain more. This variant is intended to be used when the package is managed only.